geo/src/algorithm/linestring_segment.rs

```diff
@@ -96,7 +97,9 @@ impl LineStringSegmentize for LineString {
             // update cumulative length
             cum_length += length;
 
-            if (cum_length >= segment_length) && (i != (n_lines - 1)) {
+            // if the cumulative lenght is greater than or equal to the segment length we
+            // must cut the line in the middle and push the coords
+            if (cum_length >= segment_length) || (cum_length == segment_length) {
```

Comment:
Here we remove the check for the i = n_lines to allow the last iteration to enter the if statement.